techdebt: [CDS-78554]: support secret variable type in ShellScriptStep (#3795)
harness/developer-hub: docs/continuous-delivery/x-platform-cd-features/cd-steps/utilities/shell-script-step.md

For more information, go to [add text secrets](/docs/platform/secrets/add-use-text-secrets).

There are two ways to use Secret Variables in Scripts:
* Variables of secret type can be directly used, like any other variable with secret variable identifier being passed to `value` attribute or field.

> * Secondly, you can use `<+secrets.getValue("secret_Id")>` to refer to the secret, and select the type of variable as String. With this approach, the variables is fetched in the string format, and then it gets masked.

Reviewer: With this approach, the variables is fetched in the string format, and then it gets masked. -> I think this is redundant and create even more confusion

Author: Modified the text.

> Usage of Secret variable is explained later in the documentation in detail.

Reviewer: can you please point me to this documentation ?

> There are two ways to use Secret Variables in Scripts:
> * Variables of secret type can be directly used, like any other variable with secret variable identifier being passed to `value` attribute or field.
> * Secondly, you can use `<+secrets.getValue("secret_Id")>` to refer to the secret for `value` attribute or field, and select the type of variable as String, which makes the variable to be of string format having value of Secret variable.
> Usage of Secret variable is explained later in the documentation in detail.

Reviewer 2: ```suggestion
There are two ways to use secret variables in scripts:
* Secret type variables can be used like any other variable with the secret variable identifier being passed to the `value` attribute or field.
* You can use `<+secrets.getValue("secret_Id")>` to refer to the secret `value` attribute or field, and select the type of variable as String. This formats the variable in string format and contains the value of the secret variable.
Secret variable usage is explained later in the documentation in detail.
```

For more information, go to [add text secrets](/docs/platform/secrets/add-use-text-secrets).

There are two ways to use secret variables in scripts:
* Secret type variables can be used like any other variable with the secret variable identifier being passed to the `value` attribute or field.
* You can use `<+secrets.getValue("secret_Id")>` to refer to the secret `value` attribute or field, and select the type of variable as String. This formats the variable in string format and contains the value of the secret variable.

> Below is an example on usage of Secret Variables in ShellScriptStep where `temp_secret` is a secret variable, and we try to access it.

Reviewer 2: ```suggestion
Below is an example of using secret variables in the Shell Script step where `temp_secret` is a secret variable and we try to access it.
```

```yaml
pipeline:
  stages:
    - stage:
        identifier: tmpStg
        type: Custom
        spec:
          execution:
            steps:
              - step:
                  type: ShellScript
                  identifier: tmpScript
                  spec:
                    shell: Bash
                    onDelegate: true
                    source:
                      type: Inline
                      spec:
                        script: |-
                          echo <+pipeline.stages.tmpStg.spec.execution.steps.tmpScript.spec.environmentVariables.secret_1>
                          echo <+pipeline.stages.tmpStg.spec.execution.steps.tmpScript.spec.environmentVariables.secret_2>
                    environmentVariables:
                      - name: secret_1
                        type: Secret
                        value: temp_secret
                      - name: secret_2
                        type: String
                        value: <+secrets.getValue("temp_secret")>
                    outputVariables: []
```

> Output for a pipeline like above, would look as shown below:

Reviewer 2: ```suggestion
The output for this step would look like this:
```

For more information, go to [add text secrets](/docs/platform/secrets/add-use-text-secrets).

> Basically, you use `<+secrets.getValue("secret_Id")>` to refer to the secret. Also, to access input and output variables of Secret type, you can directly use the variable, as explained later in the documentation.

Reviewer: Now, the main way of adding secrets should be using secret type variables, so that should come first. Using 
<+secrets.getValue()> should be mentioned later and we should also mention if there is any difference between their usage

Author: Modified the context, prioritising the application of the new feature.
